coa/pkg/apis/v1alpha2/bindings/http/telemetry.go

```diff
@@ -24,19 +24,24 @@ var initialized bool
 
 func initClient(properties map[string]interface{}) {
 	instrumentationKey := os.Getenv("APP_INSIGHT_KEY")
-	if instrumentationKey == "" {
-		instrumentationKey = "0be0a36e-6e0a-4544-a453-a237fd25cf64"
-	}
 	telemetryConfig := appinsights.NewTelemetryConfiguration(instrumentationKey)
```

Comment:
What happens if the key is missing?

Reply:
You are right, I have added a check in `http_pipeline.go` to validate the existence of env var - `APP_INSIGHT_KEY`. If it is missing, will return `BadConfig` error. Meanwhile, I also guard the existence of env var in initClient. If absence, will set `initialized = false`